iml-services/iml-snapshot/src/main.rs

```diff
@@ -2,16 +2,115 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
+use core::fmt::Debug;
 use futures_util::stream::TryStreamExt;
+use iml_api_utils::{wait_for_cmds_success, CommandError};
+use iml_graphql_queries::snapshot as snapshot_queries;
+use iml_manager_client::{get_influx, graphql};
 use iml_manager_env::get_pool_limit;
 use iml_postgres::{get_db_pool, sqlx};
 use iml_service_queue::service_queue::consume_data;
 use iml_tracing::tracing;
 use iml_wire_types::snapshot;
+use reqwest::Client;
+use std::{collections::HashMap, collections::HashSet, sync::Arc};
+use tokio::{
+    sync::Mutex,
+    time::Instant,
+    time::{interval, Duration},
+};
 
 // Default pool limit if not overridden by POOL_LIMIT
 const DEFAULT_POOL_LIMIT: u32 = 2;
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+struct SnapshotId {
+    filesystem_name: String,
+    snapshot_name: String,
+    snapshot_fsname: String,
```

Comment:
Technically only this is a unique snapshot id. Other thing can be stored along with the State.

Reply:
I don't think there's a guarantee for `snapshot_fsname` to be unique across filesystems.

Reply:
It has to be by definition, I think, and currently it's a random string :)

Reply:
Once you query from the db, you can use the `id` column of snapshot, which is guaranteed to be unique.